### PR TITLE
Video cloudkit

### PR DESCRIPTION
**Pull request by zarghol**
This MR includes several changes :
- Video Handling on CloudKit
- Video is imported via file importer, not yet via drag n drop
- Some UI rebalance
- Separation of Views and Logic

### CODE REVIEW

**Review by Neckow — APPROVED** [review]

File: RunningOrder/Shared/MainView+Logic.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+//
+//  Created by Clément Nonn on 19/02/2021.
+//  Copyright © 2021 Worldline. All rights reserved.
+//
+
+import SwiftUI
+import Combine
+
+extension MainView {
+    final class Logic: ObservableObject {
```
Comment (ghiitais) [inline RunningOrder/Shared/MainView+Logic.swift:13]: class name is unusual, why did you call it Logic ?

Comment (zarghol) [reply RunningOrder/Shared/MainView+Logic.swift:13]: In terms of design pattern and Architeture, we should have a separation of concerns between the logic work / business rules, and the views. In a MVVM architecture, we usually have ViewModels that contains bindings towards the associated View, and the logic of the view. Here I did another cut : I separated just the logic part from the view, not the bindings that could leverage all the power of SwiftUI with property wrappers. This way we have just Logic unit linked to their views. I am open if you find a better name :)

File: RunningOrder/Story/Extensions/StoryInformation+CloudKit.swift [hunk 2]
```diff
@@ -7,57 +7,67 @@
...
-
-    static func recordName(for storyId: String) -> String {
-        return "si-\(storyId)"
-    }
-
-    private func storyRecordId(zoneId: CKRecordZone.ID) -> CKRecord.ID {
-        return CKRecord.ID(recordName: self.storyId, zoneID: zoneId)
-    }
-}
+//import CloudKit
```
Comment (ghiitais) [inline RunningOrder/Story/Extensions/StoryInformation+CloudKit.swift]: why do you want to keep this code commented ?

File: RunningOrder/Story/Models/StoryInformation.swift [hunk 3]
```diff
@@ -15,7 +16,139 @@ struct StoryInformation {
...
+            } else if let oldUrl = oldValue,
+                      let currentExtension = videoExtension,
+                      videoUrl == nil,
+                      let symbolicUrl = try? Self.symbolicVideoUrl(videoUrl: oldUrl, videoExtension: currentExtension, fileManager: fileManager),
+                      fileManager.fileExists(atPath: symbolicUrl.path) {
+                try? fileManager.removeItem(at: symbolicUrl)
+                self.videoExtension = nil
+            }
+        }
+    }
```
Comment (ghiitais) [inline RunningOrder/Story/Models/StoryInformation.swift:34]: line break here

**Review by ghiitais — APPROVED** [review]

File: RunningOrder/Story/Extensions/StoryInformation+CloudKit.swift (hunk 2 above)
Comment (zarghol) [reply RunningOrder/Story/Extensions/StoryInformation+CloudKit.swift]: file removed

**Pull request merged**